physlr/physlr.py

```diff
@@ -2236,24 +2237,216 @@ def physlr_bed_to_path(self):
+                continue
+
+            contig1 = columns[0].rstrip("-+")
+            contig2 = columns[1].rstrip("-+")
+            dist[(contig1, contig2)] = int(columns[idx])
+            dist[(contig2, contig1)] = int(columns[idx])
+
+        dist_est_file.close()
+        print("Read Distance Estimates", file=sys.stderr)
+        return dist
+
+    @staticmethod
+    def normal_estimation(x, probability, n):
+        """
+        Compute binomial estimation
+        """
+        mean_val = n * probability
+        std_dev = float(math.sqrt(mean_val * (1 - probability)))
+        return 0.5 * (1 + math.erf((x - mean_val) / (std_dev * math.sqrt(2))))
+
+    @staticmethod
+    def is_link_significant(ori_list):
+        """
+        Check if barcode link is strong enough
+        """
+        max_idx = ori_list.index(max(ori_list))
+        ori_list.sort()
+        max_val = ori_list[-1]
+        sum_with_second_max = ori_list[-1] + ori_list[-2]
+        normal_cdf = Physlr.normal_estimation(max_val, 0.5, sum_with_second_max)
+        if 1 - normal_cdf < 0.05:
+            return max_idx
+        return -1
+
+    @staticmethod
+    def orient_part_of_path_backward(pairs, path, unoriented, curr_pos, name):
+        """
+        Orient small part of path based on ARCS scaffold pairing information going backwards
+        """
+        idxtojoin = {0:"-+", 1:"--", 2:"++", 3:"+-"}
+        while unoriented:
+            prev_pos = curr_pos - 1
+            pair = (path[prev_pos][:-1], name[:-1])
+            if pair in pairs:
+                join_ori = pairs[pair]
+                max_idx = Physlr.is_link_significant(join_ori)
+                if max_idx != -1:
+                    curr_ori = idxtojoin[max_idx][1]
+                    if curr_ori == name[-1]:
+                        prev_ori = idxtojoin[max_idx][0]
+                        path[prev_pos] = path[prev_pos][0:-1] + prev_ori
+                        del unoriented[-1]
+                        curr_pos = prev_pos
+                        name = path[prev_pos]
+                    else:
+                        unoriented.clear()
+                else:
+                    unoriented.clear()
+            else:
+                unoriented.clear()
+
+        return path, unoriented
+
+
+    @staticmethod
+    def orient_part_of_path_forward(pairs, path, unoriented, curr_pos, name):
+        """
+        Orient small part of path based on ARCS scaffold pairing information going forwards
+        """
+        idxtojoin = {0:"-+", 1:"--", 2:"++", 3:"+-"}
+        prev_pos = curr_pos - 1
+        prev_name = path[prev_pos]
+        pair = (prev_name[:-1], name[:-1])
+        if pair in pairs:
+            join_orientation = pairs[pair]
+            max_idx = Physlr.is_link_significant(join_orientation)
+            if max_idx != -1:
+                if idxtojoin[max_idx][0] == prev_name[-1]:
+                    path[curr_pos] = name[:-1] + idxtojoin[max_idx][1]
+                else:
```

Comment:
Similar to suggestion for `orient_part_of_path_backward` -- Perhaps using a flag to get rid of the duplicated code

Reply:
done